src/fragmentarium/ui/SearchForm.test.tsx

```diff
@@ -5,9 +5,20 @@ import { factory } from 'factory-girl'
 import Promise from 'bluebird'
 import SessionContext from 'auth/SessionContext'
 import SearchForms from './SearchForm'
-import { createMemoryHistory } from 'history'
+import { createMemoryHistory, MemoryHistory } from 'history'
 import BibliographyEntry from 'bibliography/domain/BibliographyEntry'
 import userEvent from '@testing-library/user-event'
+import FragmentService from 'fragmentarium/application/FragmentService'
+import FragmentSearchService from 'fragmentarium/application/FragmentSearchService'
+import MemorySession from 'auth/Session'
+
+jest.mock('fragmentarium/application/FragmentService')
+
+const fragmentService = new (FragmentService as jest.Mock<FragmentService>)()
+const fragmentSearchService = new (FragmentSearchService as jest.Mock<
+  FragmentSearchService
+>)()
+const session = new (MemorySession as jest.Mock<MemorySession>)()
```

Comment:
You could just create a real `MemorySession` as it does not require complex setup. Either pass the desired scopes or use spies.

If you want to use a class mock, remember to set it up.

Reply:
so spying like this:
```
const session = new (MemorySession as jest.Mock<MemorySession>)()
jest.spyOn(session, "isAllowed...").mockReturnValue 
```?

Trying:
```
const session = new (MemorySession as jest.Mock<<jest.Mocked<MemorySession>>)()
session.isAllowedToRead.mockReturnValue(true) 
```
errors with not a function.

But using similar pattern on fragmentService `( as jest.Mock<jested.Mock<FragmentService>>)` and then
using `fragmentService.statistics.mockReturnValue...` works fine...
Just MemorySession it errors but not on FragmentService.

Reply:
Maybe i can just not Mock the MemorySession (just instantiate it with scope). Would this be ok ? Seems like it would make it more of an integration test then. It would be the easiset way, saving some lines of code.

Reply:
i forgot to import session as `jest.mock("auth/session") now it works. But i init Session as a real instance (not a mock) which i prefer.